### PR TITLE
NCL-1157 - Running Builds and Completed Builds should use the same en…

### PR DESCRIPTION
**Pull request by vibe13**
…dpoint - latest refactoring

### CODE REVIEW

**Comment by vibe13** [comment]
@alexcreasy  @matejonnet This finally replaces https://github.com/project-ncl/pnc/pull/878 and https://github.com/project-ncl/pnc/pull/887

**Comment by vibe13** [comment]
test this please

**Comment by vibe13** [comment]
retest this please

File: rest/src/main/java/org/jboss/pnc/rest/endpoint/UserEndpoint.java [hunk 1]
```diff
@@ -193,4 +193,23 @@ public Response update(@ApiParam(value = "User id", required = true) @PathParam(
        return super.update(id, userRest);
     }
 
+    // TODO To be removed after testing, will be available via pnc-rest/rest/builds?q=(user.id==1)
```
Comment (vibe13) [inline rest/src/main/java/org/jboss/pnc/rest/endpoint/UserEndpoint.java]: @matejonnet see https://github.com/project-ncl/pnc/pull/878#discussion_r55342308

File: rest-model/src/main/java/org/jboss/pnc/rest/restmodel/BuildRecordRest.java [hunk 2]
```diff
@@ -119,9 +129,15 @@ public BuildRecordRest(BuildRecord buildRecord) {
         this.performedMilestoneBuildRecordSetIds = nullableStreamOf(buildRecord.getBuildRecordSets())
                 .filter(buildRecordSet -> buildRecordSet.getPerformedInProductMilestone() != null)
                 .map(buildRecordSet -> buildRecordSet.getId()).collect(Collectors.toSet());
+
+        performIfNotNull(buildRecord.getUser(), () -> user = new UserRest(buildRecord.getUser()));
```
Comment (vibe13) [inline rest-model/src/main/java/org/jboss/pnc/rest/restmodel/BuildRecordRest.java]: @matejonnet  simplified logic now, see https://github.com/project-ncl/pnc/pull/878#discussion_r55342003

File: rest/src/main/java/org/jboss/pnc/rest/endpoint/BuildConfigurationEndpoint.java [hunk 3]
```diff
@@ -508,4 +508,23 @@ public Response getLatestBuildRecord(@ApiParam(value = "Build configuration id",
         return this.fromSingleton(buildRecordProvider.getLatestBuildRecord(id));
     }
 
+    //TODO To be removed after testing, will be available via pnc-rest/rest/builds?q=buildConfigurationAudited.idRev.id==1
```
Comment (vibe13) [inline rest/src/main/java/org/jboss/pnc/rest/endpoint/BuildConfigurationEndpoint.java]: @matejonnet see https://github.com/project-ncl/pnc/pull/878#discussion_r55342308

File: rest-model/src/main/java/org/jboss/pnc/rest/restmodel/BuildExecutionConfigurationRest.java [hunk 4]
```diff
@@ -95,6 +95,18 @@ public BuildExecutionConfiguration toBuildExecutionConfiguration() {
         );
     }
 
+    public BuildConfigurationAuditedRest createBuildConfigurationAuditedRest() {
```
Comment (vibe13) [inline rest-model/src/main/java/org/jboss/pnc/rest/restmodel/BuildExecutionConfigurationRest.java]: @matejonnet see https://github.com/project-ncl/pnc/pull/878#discussion_r55341555. Now the relation is between 2 rest-model objects, not anymore between a rest-model object and a db entity. I could not avoid this conversion (from BuildExecutionConfigurationRest to BuildConfigurationAuditedRest), as the rsql relies on the presence of this object also for the running builds, for being able to apply the filters.

Comment (matejonnet) [inline rest-model/src/main/java/org/jboss/pnc/rest/restmodel/BuildExecutionConfigurationRest.java]: Now it aligns with other *REST DTOs. Ok to merge.
We are going to revisit this soon as we plan to decouple rest-model from db-model.

**Comment by vibe13** [comment]
@matejonnet  whooo thanks! merging!!

**Pull request merged**